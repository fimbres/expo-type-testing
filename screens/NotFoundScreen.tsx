import { FC } from 'react';
import { StyleSheet, TouchableOpacity, View, Text } from 'react-native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';

import { RootStackParamList } from '../navigation/types';

const NotFoundScreen: FC<NativeStackScreenProps<RootStackParamList, "NotFound">> = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>This product doesn't exist.</Text>
      <TouchableOpacity onPress={() => navigation.navigate('Home')} style={styles.link}>
        <Text style={styles.linkText}>Go to home screen!</Text>
      </TouchableOpacity>
    </View>
  );
}

export default NotFoundScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 20,
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  link: {
    marginTop: 15,
    paddingVertical: 15,
  },
  linkText: {
    fontSize: 14,
    color: '#2e78b7',
  },
});
