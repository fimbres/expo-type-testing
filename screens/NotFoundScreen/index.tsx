import { FC } from "react";
import { StyleSheet, View } from "react-native";
import { NativeStackScreenProps } from "@react-navigation/native-stack";

import { RootStackParamList } from "../../navigation/types";
import {
  AppButton,
  SafeAreaContainer,
  ScreenFooter,
  ScreenHeader,
} from "../../components";
import { Colors } from "../../constants";

const NotFoundScreen: FC<
  NativeStackScreenProps<RootStackParamList, "NotFound">
> = ({ navigation }) => {
  return (
    <>
      <SafeAreaContainer style={styles.statusBar} />
      <SafeAreaContainer style={{ paddingTop: 0 }}>
        <View style={styles.container}>
          <ScreenHeader
            title="Lo Sentimos!"
            subtitle="El producto que seleccionaste no existe"
            style={styles.header}
          />
          <ScreenFooter style={styles.footer}>
            <AppButton
              title="Volver a Inicio"
              size="lg"
              onPress={() =>
                navigation.navigate("Home", { initialFilter: "all" })
              }
            />
          </ScreenFooter>
        </View>
      </SafeAreaContainer>
    </>
  );
};

export default NotFoundScreen;

const styles = StyleSheet.create({
  statusBar: {
    flex: 0,
    backgroundColor: Colors.white,
  },
  container: {
    flex: 1,
    justifyContent: "space-between",
  },
  header: {
    paddingTop: 66,
    paddingBottom: 24,
    borderBottomWidth: 1,
    borderColor: Colors.black,
    backgroundColor: Colors.white,
  },
  footer: {
    padding: 20,
  },
});
