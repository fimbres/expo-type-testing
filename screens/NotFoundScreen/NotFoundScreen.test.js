import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react-native';

import NotFoundScreen from '.';

describe('<NotFoundScreen />', () => {
    it('render text correctly', () => {
        render(
            <NotFoundScreen />
        );

        const firstText = screen.getByText('Lo Sentimos!');
        const secondText = screen.getByText('El producto que seleccionaste no existe');
        const thirdText = screen.getByText('Volver a Inicio');

        expect(firstText).toBeDefined();
        expect(secondText).toBeDefined();
        expect(thirdText).toBeDefined();
    });

    it('renders correctly', () => {
        const tree = renderer
        .create(
            <NotFoundScreen />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
