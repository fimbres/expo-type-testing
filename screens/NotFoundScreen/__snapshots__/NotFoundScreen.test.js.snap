// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundScreen /> renders correctly 1`] = `
Array [
  <RCTSafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#F8F8F8",
          "flex": 1,
          "paddingTop": 0,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 0,
        },
      ]
    }
  />,
  <RCTSafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#F8F8F8",
          "flex": 1,
          "paddingTop": 0,
        },
        Object {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 20,
              "paddingTop": 24,
            },
            Object {
              "backgroundColor": "#FFFFFF",
              "borderBottomWidth": 1,
              "borderColor": "#020202",
              "paddingBottom": 24,
              "paddingTop": 66,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#020202",
              "fontFamily": "avenir-black",
              "fontSize": 20,
              "fontWeight": "800",
              "lineHeight": 27.32,
            }
          }
        >
          Lo Sentimos!
        </Text>
        <Text
          style={
            Object {
              "color": "#020202",
              "fontFamily": "avenir-book",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 21.86,
            }
          }
        >
          El producto que seleccionaste no existe
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "marginVertical": 40,
            },
            Object {
              "padding": 20,
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#334FFA",
                "borderRadius": 10,
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "paddingVertical": 14,
              },
              false,
            ]
          }
          testID="app-button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#020202",
                  "fontFamily": "avenir-black",
                  "fontSize": 16,
                  "fontWeight": "800",
                  "lineHeight": 21.86,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Volver a Inicio
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>,
]
`;
